### PR TITLE
refactor(JiraCard): update to composability api

### LINKED ISSUE #77: EntityJiraOverviewCard consumes entity via props
## Expected Behavior

Component should support new composability API by consuming the entity via context rather than props, i.e. via `useEntity`

## Current Behavior

https://github.com/RoadieHQ/backstage-plugin-jira/blob/d0a0e589e03abdaa094f6cbc179ae8a99ead01b9/src/components/JiraCard/JiraCard.tsx#L79

## Possible Solution

Use `useEntity` instead of props

### PR DESCRIPTION
**Pull request by kuangp**
Closes #77

### CODE REVIEW

**Comment by bobalong79** [comment]
Thanks for this. Can confirm it works, however when trying to find a way to change this conditional for the composability API I ran into an issue...

```ts
{isJiraAvailable(entity) && (
  <Grid item md={6}>
    <JiraCard entity={entity} />
  </Grid>
)}
```

I tried:

```ts
<EntitySwitch>
  <EntitySwitch.Case if={isJiraAvailable}>
    <Grid item md={6}>
      <EntityJiraOverviewCard />
    </Grid>
  </EntitySwitch.Case>
</EntitySwitch>
```

But got a compiler error as the `isJiraAvailable` definition is a `string | undefined` not a `boolean`:

```
Type '(entity: Entity) => string | undefined' is not assignable to type '(entity: Entity) => boolean'.
```

Changing the definition in `Router.tsx` did the trick:

```ts
export const isJiraAvailable = (entity: Entity) =>
  Boolean(entity.metadata.annotations?.[JIRA_PROJECT_KEY_ANNOTATION]);
```

Feel free to include in your PR or I can submit a separate one.

**Comment by kuangp** [comment]
@bobalong79 updated based on your suggestion, thanks!

**Comment by sjuxax** [comment]
Working well here. Thanks!

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   "dependencies": {
```
Comment (Xantier) [inline package.json:28]: @kuangp could you bump the package version to `1.0.0` with this PR?
I will merge and release with that change

Comment (kuangp) [reply package.json:28]: done! thanks!

**Review by Xantier — APPROVED** [review]
Thank you very much for the contribution @kuangp 

Merging and releasing..

**Pull request merged**